src/mongo/delegates/Updates.class.php

```diff
@@ -229,19 +235,26 @@ public function saveChanges(
     protected function setReadPreferenceToPrimary()
     {
         // Set db preference
-        $dbPref = $this->getDatabase()->getReadPreference();
-        if($dbPref['type'] !== \MongoClient::RP_PRIMARY){
-            $this->originalDbReadPreference = $this->db->getReadPreference();
-            $tagsets = (isset($dbPref['tagsets']) ? $dbPref['tagsets'] : array());
-            $this->db->setReadPreference(\MongoClient::RP_PRIMARY, $tagsets);
+        /** @var ReadPreference $dbReadPref */
+        $dbReadPref = $this->getDatabase()->__debugInfo()['readPreference'];
```

Comment:
It's not implemented currently - so at some point i imagine it will be and we can remove that. At which point we'll be bumping up a change of the driver and we'll be using new functionality. We should probably raise an issue to track that though.